docs/developer-guide/working-with-plugins.md

```diff
@@ -66,6 +66,28 @@ module.exports = {
 };
 ```
 
+### Custom Environments in Plugins
+
+Plugins can also provide custom environments. Custom environments can declare `globals`, turn `ecmaFeatures` on and off,
+and configure both core rules as well as rules included in the plugin. To provide custom environment, include `environment`
+property in your exported object.
```

Comment:
Environments can't specify plugins, only `globals`, `ecmaFeatures`, and `rules`.